userspace/libsinsp/sinsp_filtercheck_k8s.cpp

```diff
@@ -37,7 +37,9 @@ static inline bool str_match_start(const std::string& val, size_t len, const cha
 static const filtercheck_field_info sinsp_filter_check_k8s_fields[] =
 {
 	{PT_CHARBUF, EPF_NONE, PF_NA, "k8s.pod.name", "Pod Name", "Kubernetes pod name."},
-	{PT_CHARBUF, EPF_NONE, PF_NA, "k8s.pod.id", "Pod ID", "Kubernetes pod id."},
+	{PT_CHARBUF, EPF_NONE, PF_NA, "k8s.pod.id", "Pod / Sandbox ID", "The truncated Kubernetes pod / sandbox ID (first 12 characters), e.g 63060edc2d3a. Note that the pod ID is the sandbox ID in the Kubernetes control plane context, not the pod UID. The pod ID is specific to the container runtime environment. It is the equivalent of the container ID for the pod / sandbox and extracted from the Linux cgroups."},
```

Comment:
@leogr :popcorn: WDYT?

Reply:
Hey @incertum 

As discussed here, https://github.com/falcosecurity/libs/pull/1489#pullrequestreview-1755900817 all existing `k8s.pod.*` and `k8s.ns.name` fields must also be kept unchanged to avoid any possible breaking change. Even in this case, adopters may rely on the current `k8s.pod.id` value format, and since, as we know, this is likely the most used field, changing it may create unexpected consequences.

So, I'm afraid I have to disagree with changing how the current `k8s.pod.id` works.

I'd rather:
- `k8s.pod.id` as-is now
- add `k8s.pod.sandbox_id` and `k8s.pod.full_sandbox_id`

This would be easily understandable by anyone and won't introduce subtle changes.

Note: We might also add the `k8s.pod.uid` field that aliases `k8s.pod.id` (as-is now), but I don't see any real benefit. So I'd avoid doing that.

Btw, everything else SGMT, and thank you for this fix! :pray: